storage/remote/dial_context.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"context"
+	"math/rand"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/common/config"
+)
+
+type hostResolver interface {
+	LookupHost(context.Context, string) ([]string, error)
+}
+
+// dialContextWithRoundRobinDNS is needed to facilitate testing, because
```

Comment:
It is used in `storage/remote/client.go`; I think you mean the members could be hard-coded for non-test code, but they are variable so that tests can change them.  Is that right?

Reply:
Exactly. Should I re-phrase that?

Reply:
I have addressed it in 7ad34a8bf1ba77335bea83fc6eed2bc85226cad6.